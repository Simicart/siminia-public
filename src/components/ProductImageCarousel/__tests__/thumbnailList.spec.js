import React from 'react';
import testRenderer from 'react-test-renderer';

import ThumbnailList from '../thumbnailList';

jest.mock('src/classify');

const updateActiveItemIndexMock = jest.fn();
const defaultProps = {
    activeItemIndex: 0,
    items: [
        {
            file: 'thumbnail1.png',
            label: 'test-thumbnail1'
        },
        {
            file: 'thumbnail2.png',
            label: 'test-thumbnail2'
        }
    ],
    updateActiveItemIndex: updateActiveItemIndexMock
};

test('renders the ThumbnailList component correctly', () => {
    const component = testRenderer.create(<ThumbnailList {...defaultProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('sets isActive on active thumbnail based on activeItemIndex', () => {
    const component = testRenderer.create(<ThumbnailList {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree.children[defaultProps.activeItemIndex].props.className).toEqual(
        'rootSelected'
    );
});

test('calls updateActiveItemIndex with index of clicked thumbnail', () => {
    const component = testRenderer.create(<ThumbnailList {...defaultProps} />);

    const buttons = component.root.findAllByType('button');
    const inactiveThumbnail = buttons.find(
        button => button.props.className === 'root'
    );

    inactiveThumbnail.props.onClick();

    expect(updateActiveItemIndexMock).toHaveBeenCalledWith(1);

    updateActiveItemIndexMock.mockReset();
});
