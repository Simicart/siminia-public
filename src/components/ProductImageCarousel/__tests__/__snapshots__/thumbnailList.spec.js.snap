// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ThumbnailList component correctly 1`] = `
<div
  className="root"
>
  <button
    className="rootSelected"
    onClick={[Function]}
  >
    <img
      alt="test-thumbnail1"
      className="image"
      src="/img/resize/240?url=%2Fmedia%2Fcatalog%2Fproduct%2Fthumbnail1.png"
    />
  </button>
  <button
    className="root"
    onClick={[Function]}
  >
    <img
      alt="test-thumbnail2"
      className="image"
      src="/img/resize/240?url=%2Fmedia%2Fcatalog%2Fproduct%2Fthumbnail2.png"
    />
  </button>
</div>
`;
